client/starwhale/core/dataset/copy.py

```diff
@@ -168,12 +172,13 @@ def _do_dataset_copy(
                 ":cookie: update rows and copy artifacts...", total=None
             )
 
+            row_updater_exceptions: t.List[Exception] = []
             row_updater_threads = []
             for i in range(0, self._workers):
                 _t = threading.Thread(
                     name=f"row-updater-{i}",
                     target=self._do_row_update,
-                    args=(progress, task_id, dest),
+                    args=(progress, task_id, dest, row_updater_exceptions),
```

Comment:
Are there has a way to fail fast, such as to define a callback function which deal with the exception, like this:
`def exception_callback(e: Exception) -> None:`
`  raise e`
so that the main thread can fail fast as soon as.

Reply:
Python does not support thread callback yet.